File: pset 2/readibility/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int letters = 0;
int words = 0;
int sentences = 0;
float L = 0;
float S = 0;
float index;
int rounded;

int main(void)
{
    char *text = get_string("Text: ");
    int len = strlen(text);

    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
        if (isspace(text[i]))
        {
            words += 1;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
                sentences += 1;
        }
    }
    words += 1;
    L = ((float) letters / ((float) words)) * 100;
    S = ((float) sentences / ((float) words)) * 100;
    index = 0.0588 * L - 0.296 * S - 15.8;
    rounded = round(index);

    if (rounded < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (rounded >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %i\n", rounded);
}

File: pset 6/sentimental-readability/readability.py
from cs50 import get_string

text_input = get_string("Text: ")

# used to convert input into  a list, split oop returs list
text = text_input.split()
letter_count = 0
sentence_count = 0

# len on list counts no of elements in it
word_count = len(text)

for letters in text_input:
    # checks if it is alphabet, could b capital or small
    if (letters.isalpha() == True):
        letter_count += 1
    if (letters == "." or letters == "?" or letters == "!"):
        sentence_count += 1


L = letter_count / word_count * 100
S = sentence_count / word_count * 100

grade = 0.0588 * L - 0.296 * S - 15.8
grade = round(grade)

if (grade < 1):
    print("Before Grade 1")

elif (grade >= 16):
    print("Grade 16+")

else:
    print(f"Grade {grade}")
